js/views/modals/listingDetail/Listing.js

```diff
@@ -103,6 +120,67 @@ export default class extends BaseModal {
     }
   }
 
+  onClickGotoPhotos() {
+    this.gotoPhotos();
+  }
+
+  gotoPhotos() {
+    this.$el.animate({
+      scrollTop: this.$photoSection.offset().top,
+    }, 500);
+  }
+
+  onClickPhotoSelect(e) {
+    this.setSelectedPhoto($(e.target).data('index'));
+  }
+
+  setSelectedPhoto(photoIndex) {
+    if (!photoIndex) {
+      throw new Error('Please provide an index for the selected photo.');
+    }
+    const photoCol = this.model.get('listing').toJSON().item.images;
+    const photoHash = photoCol[photoIndex].original;
+    const phSrc = app.getServerUrl(`ipfs/${photoHash}`);
+
+    this.$photoSection.cropit('imageSrc', phSrc);
+    this.$photoSelected.attr('data-index', photoIndex);
+  }
+
+  setActivePhotoThumbnail(thumbIndex) {
+    if (!thumbIndex) {
+      throw new Error('Please provide an index for the selected photo thumbnail.');
+    }
+    this.$photoRadioBtns.prop('checked', false).eq(thumbIndex).prop('checked', true);
+  }
+
+  onClickPhotoPrev() {
+    let targetIndex = parseInt(this.$photoSelected.attr('data-index'), 10) - 1;
+    const imagesLength = parseInt(this.model.get('listing').toJSON().item.images.length, 10);
+
+    targetIndex = targetIndex < 0 ? imagesLength - 1 : targetIndex;
+    this.setSelectedPhoto(targetIndex);
+    this.setActivePhotoThumbnail(targetIndex);
+  }
+
+  onClickPhotoNext() {
+    let targetIndex = parseInt(this.$photoSelected.attr('data-index'), 10) + 1;
+    const imagesLength = parseInt(this.model.get('listing').toJSON().item.images.length, 10);
+
+    targetIndex = targetIndex >= imagesLength ? 0 : targetIndex;
+    this.setSelectedPhoto(targetIndex);
+    this.setActivePhotoThumbnail(targetIndex);
+  }
+
+  onClickFreeShippingLabel() {
+    this.gotoShippingOptions();
+  }
+
+  gotoShippingOptions() {
+    this.$el.animate({
```

Comment:
Absolutely no need to change this, but as an FYI, I did bring [Velocity](http://velocityjs.org/) into the codebase and am using it in some places. There is a utility module to pull it in:

```
import '../../../utils/velocity';
```

I also have one for the additional [UI pack](http://velocityjs.org/#uiPack), which has some cool pre-packaged effects available:

```
import '../../../utils/velocityUiPack.js';
```

Anyways, Velocity gives much smoother animations than jQuery and has a dev friendly API.